File: app/src/main/java/pl/piotrskiba/angularowo/models/DetailedPlayer.java
package pl.piotrskiba.angularowo.models;

public class DetailedPlayer {
    private final String username;
    private final String uuid;
    private final String status;
    private final String rank;
    private final boolean vanished;
    private final float balance;
    private final int playtime;
    private final int tokens;

    public DetailedPlayer(String username, String uuid, String status, String rank, boolean vanished, float balance, int playtime, int tokens){
        this.username = username;
        this.uuid = uuid;
        this.status = status;
        this.rank = rank;
        this.vanished = vanished;
        this.balance = balance;
        this.playtime = playtime;
        this.tokens = tokens;
    }

    public String getUsername(){
        return username;
    }

    public String getUuid(){
        return uuid;
    }

    public String getStatus() {
        return status;
    }

    public String getRank(){
        return rank;
    }

    public boolean isVanished(){
        return vanished;
    }

    public float getBalance() {
        return balance;
    }

    public int getPlaytime() {
        return playtime;
    }

    public int getTokens() {
        return tokens;
    }
}

File: app/src/main/java/pl/piotrskiba/angularowo/models/DetailedPlayer.kt
package pl.piotrskiba.angularowo.models

import com.google.gson.annotations.SerializedName
import pl.piotrskiba.angularowo.utils.RankUtils
import java.io.Serializable

data class DetailedPlayer(
        val uuid: String,
        val skinUuid: String,
        val partnerUuid: String?,
        val username: String,
        val status: String,
        @SerializedName(value = "rank") val rankKey: String,
        @SerializedName(value = "vanished") val isVanished: Boolean,
        val balance: Float,
        val playtime: Int,
        val tokens: Int,
        private val permissions: List<String>
): Serializable {

    val rank: Rank
        get() = RankUtils.getRankFromName(rankKey)

    fun hasPermission(permission: String) = permissions.contains(permission)

}
